### PR TITLE
feat: add fake migrations running and reverting

### PR DESCRIPTION
**Pull request by ahmedosama94**
Added a cli option to fake-run or fake-revert a migration, adding to the
executed migrations table, but not actually running it. This feature is
useful for when migrations are added after the fact or for
interoperability between applications which are desired to each keep
a consistent migration history

Closes: #6195

### Description of change

Extended cli `migration:run` and `migration:revert` to have a `--fake` (alias `-f`) option, to fake run migrations, for migrations added after the fact (similar to Django's `python manage.py migrate --fake`). Detailed description in linked issue.

### Pull-Request Checklist

- [x] Code is up-to-date with the `master` branch
- [x] `npm run format` to apply prettier formatting
- [x] `npm run test` passes with this change ---> SAP skipped as tests threw errors that blocked all tests from being run. Some tests where also failing before the update as well (schema builder > create table), these ones do still fail (at least on my end).
- [x] This pull request links relevant issues as `Fixes #0000`
- [x] There are new or updated unit tests validating the change
- [x] Documentation has been updated to reflect this change
- [x] The new commits follow conventions explained in [CONTRIBUTING.md](https://github.com/typeorm/typeorm/blob/master/CONTRIBUTING.md)

### CODE REVIEW

File: src/migration/MigrationExecutor.ts [hunk 1]
```diff
@@ -23,6 +23,7 @@ export class MigrationExecutor {
      *   each: each migration is run in a separate transaction
      */
     transaction: "all" | "none" | "each" = "all"
+    fake: boolean
```
Comment (pleerock) [inline src/migration/MigrationExecutor.ts:26]: can you please add a description about this feature here as well?

File: test/github-issues/6195/issue-6195.ts [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+    )
+}
+
+describe("github issues > #6195 feature: fake migrations for existing tables", () => {
+    let dataSources: DataSource[]
+
+    before(async () => {
+        dataSources = await createTestingConnections({
+            entities: [__dirname + "/entity/*{.js,.ts}"],
+            enabledDrivers: [
```
Comment (pleerock) [inline test/github-issues/6195/issue-6195.ts:40]: why did you explicitly specified all the drivers? it's not scalable, if you want to ignore only one or two drivers - better to add checks in the tests like `if (dataSource.options.type === "mongodb") return`

**Comment by pleerock** [comment]
Thank you for contribution! Can you please address the given comments and resolve the conflicts before I can merge it? Thank you.

**Comment by AlexMesser** [comment]
thank you for contribution!

**Pull request merged**